edge/test/integration/device/device_test.go

```diff
@@ -31,6 +32,37 @@ import (
 	. "github.com/onsi/gomega"
 )
 
+//getting Devicestate from subscribed MQTT topic
+var DeviceState string
+
+//getting Device_state field from subscribed MQTT topic,Unmarshalling into structure variable
+
+type DeviceUpdates struct {
+	EventId     string `json:"event_id"`
+	Timestamp   int    `json:"timestamp"`
+	DeviceField `json:"device"`
+}
+type DeviceField struct {
+	Name       string `json:"name"`
+	State      string `json:"state"`
+	LastOnline string `json:"last_online"`
+}
+
+// Mqtt client is subscribed to this topic,for the state update
+func SubMessageReceived(client MQTT.Client, message MQTT.Message) {
```

Comment:
why are we passing client here ?

Reply:
SubMessagereceived is a type  of message handler function which receives two paramaeter (client,message.)..Eventhough we have not uses client inside the function defination .We should pass it otherwise compilation fails.